Tests/Common/SimulatedDevice.cs

```diff
@@ -265,73 +264,90 @@ private bool HandleJoinAccept(LoRaPayloadJoinAccept payload)
+                        {
+                            var joinAccept = new LoRaPayloadJoinAccept(StringToByteArray(somePdu), AppKey.Value);
+                            tcs.SetResult(HandleJoinAccept(joinAccept));
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        tcs.SetException(ex);
                     }
                 }
-                catch (Exception ex)
-                {
-                    this.logger.LogError(ex, "Join failed due to: '{Message}'.", ex.Message);
-                    joinSuccessful = false;
-                    joinCompleted.Release();
-                    throw;
-                }
-            }
 
-            foreach (var basicsStation in this.simulatedBasicsStations)
-                basicsStation.MessageReceived += OnMessageReceived;
+                foreach (var basicsStation in this.simulatedBasicsStations)
+                    basicsStation.MessageReceived += OnMessageReceived;
 
-            foreach (var basicsStation in this.simulatedBasicsStations)
-            {
-                await basicsStation.SerializeAndSendMessageAsync(new
+                try
                 {
-                    JoinEui = joinRequestPayload.AppEui.ToString("D", null),
-                    msgtype = "jreq",
-                    DevEui = joinRequestPayload.DevEUI.ToString("D", null),
-                    DevNonce = joinRequestPayload.DevNonce.AsUInt16,
-                    MHdr = uint.Parse(joinRequestPayload.MHdr.ToString(), NumberStyles.HexNumber, CultureInfo.InvariantCulture),
-                    MIC = joinRequestPayload.Mic.Value.AsInt32,
-                    DR = joinRequest.RadioMetadata.DataRate,
-                    Freq = joinRequest.RadioMetadata.Frequency.AsUInt64,
-                    upinfo = new
+                    foreach (var basicsStation in this.simulatedBasicsStations)
                     {
-                        gpstime = joinRequest.RadioMetadata.UpInfo.GpsTime,
-                        rctx = 10,
-                        rssi = joinRequest.RadioMetadata.UpInfo.ReceivedSignalStrengthIndication,
-                        xtime = joinRequest.RadioMetadata.UpInfo.Xtime,
-                        snr = joinRequest.RadioMetadata.UpInfo.SignalNoiseRatio
+                        await basicsStation.SerializeAndSendMessageAsync(new
+                        {
+                            JoinEui = joinRequestPayload.AppEui.ToString("D", null),
+                            msgtype = "jreq",
+                            DevEui = joinRequestPayload.DevEUI.ToString("D", null),
+                            DevNonce = joinRequestPayload.DevNonce.AsUInt16,
+                            MHdr = uint.Parse(joinRequestPayload.MHdr.ToString(), NumberStyles.HexNumber,
+                                              CultureInfo.InvariantCulture),
```

Comment:
Simpler to cast to `byte`:

```suggestion
                            MHdr = (byte)joinRequestPayload.MHdr,
```